### ISSUE TITLE
Unused variable complains about function types

### ISSUE THREAD

**Issue by leecannon** [zigtools/zls#535]
It is valid to have parameter names in function types e.g. `a_field: fn(param: bool) void,` but zls considers `param` to be unused.

This might be a duplicate of #519 but I wasn't 100% clear whether that issue was variables being considered used when they weren't or (like this) considered unused incorrectly, if it is a duplicate go ahead and close this.

**Comment by SuperAuguste** [comment]
This isn't a duplicate, no worries! I solved this issue for externs but yeah, this is definitely something that needs fixing. :)

**Comment by kcbanner** [comment]
```
pub const TestStruct = struct {
    init: fn (some_param: u32) void
};
```

Minimal test case to trigger this

**Comment by leecannon** [comment]
Fixed in #585
